### ISSUE TITLE
[Responsiveness] Continous chat

### ISSUE THREAD

**Issue by JorenC** [zond/dipact#59]
Currently it's hard to follow the conversation, because the order of chat messages is "random". Actually, it's alphabetical, but this is not clear to the end user. 

1) Add a "unread message" counter next to the chat icon in the app bar. This is always visible while game.js is active, so it will indicate a new message has been received (show or +1) and the user can act (e.g. while looking at the map waiting for a response). 
If the new message received is one in the current active channel while the chat tab is active (so currently on screen), do not add this to the counter.

2) Sort the channels by last message received or sent. This is the default behaviour in any chat icon, making sure the relevant and active conversations are on top, and also that you can also find any new messages sent at the top of the chat list. 
There would be a design reason to keep "Everyone" on top always (as a "sticky" element, because it's the most important and more "social happening" chat. So you would want to keep this on top to stimulate people using it, and creating more emotional connection with the game as a "socializing/hang out" place), although there are also are design reasons not to (breaks the flow, needs to code exceptions, if not used wastes top space). I personally would like to keep it on the top, for the reasons described above.
3) IF we have control/are instant with our GCM push message and the new message received is NOT in the currently active chat (see point 1), then show the notification. This would be an "easy" way to navigate to that message (tap the notification). However, I think we could disregard this one as there are some delay/timing issues with the notification.

**Comment by zond** [comment]
Also, I need to remember to check what's possible using https://firebase.google.com/docs/reference/js/firebase.messaging.Messaging#setBackgroundMessageHandler

**Comment by zond** [comment]
Check now, more work has been done.

**Comment by zond** [comment]
And now, even more :D

**Comment by JorenC** [comment]
Looking good, much, much more responsive. 

You can close this one, or we can keep it open until we can sort by last message received.
IF you work on that, maybe also expose the last message and sender? I'm not sure we'll need to use it, but it might be easy to open it up and use it if we want to later. I'll keep you to be the judge of that.

Alternatively, close this one and make a new one for the sorting of channels

**Comment by zond** [comment]
It sorts on last message now. I could update it to include sender and body of last message of course :)

**Comment by JorenC** [comment]
Aaaah, nice!

The lack of feedback made it unclear, that's what would help in making it
apparent that it's sorting on this.

If you can expose it, awesome, but maybe wait until I repair the design of
the chat channels.

Tonight I hope to push the orders screen next iteration, then I'll focus on
the chat messages/heads

**Comment by zond** [comment]
That's fine, I'm rolling it out now - it was just "store the entire latest message instead of just the timestamp of it", so no sweat :)

**Comment by zond** [comment]
Let's close this - everything is done I think.
